src/cljs/rems/spa.cljs

```diff
@@ -332,11 +335,16 @@
 
 (defn page []
   (let [page-id @(rf/subscribe [:page])
-        grab-focus? @(rf/subscribe [::grab-focus?])]
+        grab-focus? @(rf/subscribe [::grab-focus?])
+        theme @(rf/subscribe [:theme])
+        lang @(rf/subscribe [:user-language])]
     (lazy-load-data!)
     [:div
      [nav/navigation-widget]
-     [logo]
+     (when (or (= page-id :home)
+               (and (not ((keyword (str "navbar-logo-name-" (name lang))) theme))
+                    (not (:navbar-logo-name theme))))
+       [nav/logo])
```

Comment:
This logic is still wrong. The regular logo should be there always when configured so. It should not depend on the page id or navbar logo.

Reply:
what do you refer to as the regular logo? the hero logo element?

Reply:
<img width="689" alt="Screenshot 2021-01-26 at 16 36 49" src="https://user-images.githubusercontent.com/7344718/105859044-ba77bb80-5ff4-11eb-9d33-c426b53ce24c.png">

Reply:
Well my thinking is that ideally you can choose only navbar logo, only hero logo, both or none. That is orthogonal, flexible and the features are not coupled. I can accept the navbar logo overriding the hero logo too, but in that case I'd like to see the helper fn for this navbar logo feature detection (that is duplicated above in the navbar).

Reply:
I thought we agreed on the spec already many revisions of this PR ago:

- Home page always shows the hero image
- Other pages don't show the hero image if the navbar image is defined

If the options were just navbar, hero, both or none, you couldn't have the hero just on the front pages, which _I think_ we imagine most users will want.

So let's finish this PR with that spec, and discuss changes to the spec afterwards, right?

Reply:
Unfortunately I think we didn't have an agreement after all but we can live with this interpretation as long as the bug is fixed.

Reply:
Bug as in the language sub.

Reply:
Isn`t this what I did @opqdonut ?
I mean I had the same thought, that we reaches the understanding like you described

Reply:
https://github.com/CSCfi/rems/pull/2483/files#r564566475 
I just went through the entitre messaging history and I swear I cannot find a single place where you would suggest this specific pattern

Reply:
@marharyta  yeah, I was replying to @Macroz

Reply:
We obviously talked past each other in this. The idea for this feature was pretty clear to me in the beginning already but all the work around the various issues is enough to confuse and miss the details. This specific solution fulfills the main point (that it does not break backwards compatibility) so like I said we can go on with this as long as the bug is fixed.

![hero_or_navbar](https://user-images.githubusercontent.com/823661/105875505-8c02dc00-6006-11eb-8a26-7603038d843b.png)

Reply:
Pretty much yes, it is implied. If two features are completely separate they are said to be orthogonal. They can be both turned on or off independently i.e. they can be only one, only other, both or none. An ideal feature is orthogonal from others, otherwise it is said to be coupled (and cannot be understood separately). For a white-label product such as REMS, I think it is desirable to have a set of features that work independently and flexibly, because the needs of different installations vary and we cannot predict everything. This is a tiny case in that sense but I think the principle is valuable overall in software development.

Reply:
ok, so how would that functionality align with the implementation?
most of out users have links to hero image in the theme, so if the rules work as true for the hero, it will always be false for the nav logo. Are you implying there should also be an extra setting for turn nav logo on? Turn hero off? a combo?

Reply:
The idea was about both of the features having their own configs to turn them on separately. For the specific case of navbar logo but only front page hero logo there needs to be a new config or solution though but we don't need to go there now. Like I said this logic is fine now because it is backwards compatible and the improvement would be tiny.